packages/playback-core/test/player.test.js

```diff
@@ -0,0 +1,169 @@
+import { assert, aTimeout } from '@open-wc/testing';
+import { initialize, teardown, MediaError, getError } from '../src/index.ts';
+
+describe('playback core', () => {
+  it('initialize w/ default preferPlayback', async function () {
+    const video = document.createElement('video');
+    const core = initialize(
+      {
+        src: 'https://stream.mux.com/23s11nz72DsoN657h4314PjKKjsF2JG33eBQQt6B95I.m3u8',
+      },
+      video
+    );
+    assert.equal(typeof core, 'object');
+    assert.equal(typeof core.engine, 'object');
+  });
+
+  it('teardown', async function () {
+    const video = document.createElement('video');
+    const core = initialize(
+      {
+        src: 'https://stream.mux.com/23s11nz72DsoN657h4314PjKKjsF2JG33eBQQt6B95I.m3u8',
+      },
+      video
+    );
+    teardown(video, core);
+
+    assert(!video.hasAttribute('src'), 'src is removed on teardown');
+  });
+
+  it('handle error', async function () {
+    const video = document.createElement('video');
+    initialize(
+      {
+        src: 'https://mux.com/', // not working src url
+      },
+      video
+    );
+    // error event must be attached after initialize(), otherwise the
+    // native error will not be stopped propagating in playback core.
+    await new Promise((resolve) => video.addEventListener('error', resolve));
```

Comment:
suggestion(non-blocking): This comment isn't quite clear (at least for me). I think you mean something like:

> Wait for the error event event to be dispatched before checking the error state

Reply:
this actually has to do with the order of error event listeners being added to the media element because the playback core stops propagation of the native error event. I'll check if we can make this better.